Makefile.dep

```diff
@@ -1038,6 +1036,64 @@ ifneq (,$(filter devfs_random,$(USEMODULE)))
   USEMODULE += random
 endif
 
+ifneq (,$(filter auto_init,$(USEMODULE)))
```

Comment:
I think it would make sense to turn this into a loop.
Something like

```
HAS_AUTO_INIT += foo \ 
                                  bar \
                                  #
ifneq (,$(filter auto_init,$(USEMODULE)))
 DEFAULT_MODULE += $(patsubst %,auto_init_%,$(filter $(HAS_AUTO_INIT),$(USEMODULE))
endif
```

Reply:
maybe additionally filter on `DISABLED_MODULES`

Reply:
> ```
> HAS_AUTO_INIT += foo \ 
>                  bar \
> ```

where would this be?

Reply:
I think in `Makefile.dep` for now is fine, no?

Reply:
If we add something like ```if not auto_init in usemodule: filter-out(auto_init_%, DEFAULT_MODULE)``` near the end of the Makefile.dep, we can ssave the repeated ```ifneq (,$(filter auto_init,$(USEMODULE)))``` for each module.

Reply:
> ```
> ifneq (,$(filter foo,$(USEMODULE)))
>   ifneq (,$(filter auto_init,$(USEMODULE)))
>     DEFAULT_MODULE += auto_init_foo
>   endif
> endif
> ```

This is what I had initially and changed to a suggestion from @aabadie 

> Instead of pulling this by default here why not rather add the following later in the file:
> 
> ```makefile
> ifneq (,$(filter auto_init,$(USEMODULE)))
>   ifneq (,$(filter gnrc_uhcpc,$(USEMODULE)))
>     DEFAULT_MODULE += auto_init_gnrc_uhcpc
>   endif
> 
>   [... other auto_init_% default modules] 
> endif
> ```
> 
> ?

Reply:
I don't like the `HAS_AUTO_INIT` because these function do not allways match a module name which makes it kind of weird. These `HAS_AUTO_INIT` would IMO have to be in the modules `MAKEFILE`, but some of these `MODULES` are not pseudomodules, so it isn't a `HAS_AUTO_INIT` but use this `MODULE`s `init` function in a way, and there isn't a `MODULE`<->`auto_init` naming match in all cases.

Reply:
> This is what I had initially and changed to a suggestion from @aabadie

hmok, @aabadie, why "later in the file"?

Reply:
>  why "later in the file"?

in order to know all modules with potential auto_init support. but since this file included transitively, this doesn't make much sense.

Reply:
So which style is prefered @aabadie @kaspar030?

Reply:
For me,
```
ifneq (,$(filter foo,$(USEMODULE)))
    DEFAULT_MODULE += auto_init_foo
   ...
endif
```

... together with some logic so that if auto_init is disabled, all auto_init_% are also disabled.